src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +20,56 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object classImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                classImplementationInstance = createNewInstance(clazz);
+                field.setAccessible(true);
+                try {
+                    field.set(classImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialise field. Class: "
+                            + clazz.getName()
+                            + ". Field: "
+                            + field.getName());
+                }
+            }
+        }
+        if (classImplementationInstance == null) {
+            classImplementationInstance = createNewInstance(clazz);
+        }
+        return classImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        Constructor<?> constructor;
+        Object instance;
+        try {
+            constructor = clazz.getConstructor();
+            instance = constructor.newInstance();
+        } catch (InstantiationException
+                | NoSuchMethodException
+                | InvocationTargetException
+                | IllegalAccessException e) {
```

Comment:
```suggestion
        } catch (ReflectiveOperationException e) {
```
we can use this for simplicity